realm/realm-library/src/main/java/io/realm/internal/OsResults.java

```diff
@@ -352,8 +352,14 @@ public Object getValue(int index) {
         return nativeGetValue(nativePtr, index);
     }
 
+    @Nullable
     public UncheckedRow getUncheckedRow(int index) {
-        return table.getUncheckedRowByPointer(nativeGetRow(nativePtr, index));
+        long rowPointer = nativeGetRow(nativePtr, index);
+        if (rowPointer == -1) {
```

Comment:
Maybe move this magical value to a constant. That way it can also be shared on the native side, see e.g `SyncSession.STATE_VALUE_ACTIVE` and the check in `SyncSession.cpp`